### PR TITLE
I3 fmt fix

### PR DESCRIPTION
**Pull request by philippeller**
i3 frame format changes to better interface with oscNext processing:
* single numeric types stores at `I3Double`, `I3Int` and `I3Bool`
* `fit_status` always populated

### CODE REVIEW

File: retro/reco.py [hunk 1]
```diff
@@ -419,23 +427,23 @@ def _reco_event(self, event, method, save_llhp, filter, save_estimate=True):
...
+            return FitStatus.InsufficientHits.value
 
         if len(event["hits"]) < MIN_NUM_HITS:
             print(
                 "fewer than {} hits found; skipping event (index {})".format(
                     MIN_NUM_HITS, event.meta["event_idx"]
                 )
             )
-            return -1
+            return FitStatus.Skipped.value
```
Comment (jllanfranchi) [inline retro/reco.py]: ditto

File: retro/reco.py [hunk 2]
```diff
@@ -419,23 +427,23 @@ def _reco_event(self, event, method, save_llhp, filter, save_estimate=True):
...
+        print("Running {} reconstruction on event".format(method))
 
         if filter and not eval(filter):  # pylint: disable=eval-used
             print(
                 "filter evaluates to False; skipping event (index {})".format(
                     event.meta["event_idx"]
                 )
             )
-            return -1
+            return FitStatus.InsufficientHits.value
```
Comment (jllanfranchi) [inline retro/reco.py]: Why are you returning `.value` when you can simply return the `IntEnum`? (which behaves as an `IntEnum` for things that know how to deal with it, and unless I'm mistaken, behaves as an `int` for things that only know how to work with integers)

File: retro/reco.py [hunk 3]
```diff
@@ -1686,15 +1694,15 @@ def func(x):
         fit_meta = OrderedDict(
             [
                 ("fit_status", np.int8(fit_status)),
-                ("iterations", np.uint32(iter_num)),
+                ("iterations", np.int32(iter_num)),
                 ("stopping_flag", np.int8(stopping_flag)),
                 ("llh_std", np.float32(fit['meta']['fstd'])),
-                ("no_improvement_counter", np.uint32(fit['meta']['no_improvement_counter'])),
+                ("no_improvement_counter", np.int32(fit['meta']['no_improvement_counter'])),
```
Comment (jllanfranchi) [inline retro/reco.py]: why the changes in types here? (it seems fine to make the change, but I'm just curious if there is a negative "special" value or something somewhere that necessitates int vs. uint)

File: retro/retro_types.py [hunk 4]
```diff
@@ -682,6 +682,7 @@ class FitStatus(enum.IntEnum):
     FailedToConverge = 30
     MissingSeed = 40
     InsufficientQuality = 50
+    Skipped = 60
```
Comment (jllanfranchi) [inline retro/retro_types.py]: Note that I created my own arbitrary code `PositiveLLH = 1` since it seems all IceCube codes are divisible by 10. Unless we can force this into IceCube codebase, maybe someone comes up with their own code in parallel, and I'm betting it's going to be divisible by 10. So I'd recommend (a) make the code e.g. 2 and (b) make sure to add a comment as I did above that it is a custom code for Retro not present in IceCube/icetray software.

File: retro/reco.py (hunk 2 above)
Comment (philippeller) [reply retro/reco.py]: I first tried to return `FitStatus.InsufficientHits` and it did not like it, so I had to change it to `.value`

File: retro/reco.py (hunk 3 above)
Comment (philippeller) [reply retro/reco.py]: there is the value `-1` is it is not initialized, and i saw it then creates overflow. Since with the new i3 format we will anyway convert to signed int, I thought i do it here directly instead of int->uint->int

File: retro/retro_types.py (hunk 4 above)
Comment (philippeller) [reply retro/retro_types.py]: ah, so the 10, 20, ... ones are I3 things? I'll change to 2 then

Comment (jllanfranchi) [reply retro/retro_types.py]: That's correct. I tried to keep fidelity to I3 software wherever I could, so all the tuples and codes should be taken directly from there, unless otherwise noted. I tried to note in docstrings where I did get things from I3 software; in this case: `Scraped from dataclasses/public/dataclasses/physics/I3Particle.h, 2019-02-18`

File: retro/reco.py (hunk 3 above)
Comment (jllanfranchi) [reply retro/reco.py]: Very well, glad you noticed that.

File: retro/reco.py (hunk 1 above)
Comment (philippeller) [reply retro/reco.py]: see below
